### PR TITLE
[Durable] Custom orchestration status

### PR DESCRIPTION
**Pull request by AnatoliB**
Feature documentation: [Custom orchestration status](https://docs.microsoft.com/azure/azure-functions/durable/durable-functions-custom-orchestration-status).

Usage sample:
``` PowerShell
Set-DurableCustomStatus -CustomStatus 'Processing'
```

### CODE REVIEW

File: src/Durable/Commands/SetDurableCustomStatusCommand.cs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+{
+    using System.Collections;
+    using System.Management.Automation;
+
+    [Cmdlet("Set", "DurableCustomStatus")]
+    public class SetDurableCustomStatusCommand : PSCmdlet
+    {
+        [Parameter(
+            Mandatory = true,
+            Position = 0,
```
Comment (Francisco-Gamino) [inline src/Durable/Commands/SetDurableCustomStatusCommand.cs:18]: Do we need to assign a `Position` for this parameter? I took a look at the other Durable cmdlets and this is the only one with a Positional parameter.

File: src/Durable/Commands/SetDurableCustomStatusCommand.cs [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+            ValueFromPipeline = true)]
```
Comment (Francisco-Gamino) [inline src/Durable/Commands/SetDurableCustomStatusCommand.cs:19]: What would the use case be here for `ValueFromPipeline`? Something like: 

```powershell
# 1)
'Processing Tokyo' | Set-DurableCustomStatus

# 2)
 @{ ProgressMessage = 'Processing Seattle'; Stage = 2 }  | Set-DurableCustomStatus

```

**Review by Francisco-Gamino — APPROVED** [review]
One minor comment and one question. Other than that, LGTM.

File: src/Durable/Commands/SetDurableCustomStatusCommand.cs (hunk 1 above)
Comment (AnatoliB) [reply src/Durable/Commands/SetDurableCustomStatusCommand.cs:18]: I'm trying to be very careful with adding positional parameters in general, but in this case I believe this is justified. The cmdlet takes just one parameter, and the purpose of this parameter is very obvious, so I want to allow users omit the parameter name: `Set-DurableCustomStatus "my status"`.

I think we can carefully allow _some_ positional parameters for _some_ other durable-related cmdlets, but we can do this later.

File: src/Durable/Commands/SetDurableCustomStatusCommand.cs (hunk 2 above)
Comment (AnatoliB) [reply src/Durable/Commands/SetDurableCustomStatusCommand.cs:19]: Yes, exactly.

File: src/Durable/Commands/SetDurableCustomStatusCommand.cs (hunk 1 above)
Comment (Francisco-Gamino) [reply src/Durable/Commands/SetDurableCustomStatusCommand.cs:18]: Sounds good @AnatoliB, thank you.

**Pull request merged**